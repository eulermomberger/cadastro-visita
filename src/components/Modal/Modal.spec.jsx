import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';

import { Modal } from '.';

const mockAddDocFunc = jest.fn();

jest.mock('firebase/firestore/lite', () => {
  return {
    addDoc() {
      mockAddDocFunc();
      return null;
    },
    collection() {
      return null;
    },
    deleteDoc() {},
    doc() { return null; },
    getDoc() {
      return {
        data() {
          return {
            name: 'Luis',
            rg: '0192839382',
            updated_at: {
              toDate() { return new Date('2023-09-24'); },
            },
          };
        },
        uuid: 'abcd123',
      };
    },
    serverTimestamp() { return new Date('2023-09-24'); },
    updateDoc() {},
    getFirestore() {},
  };
});

describe('Modal component', () => {
  it('Should renders correctly', () => {
    const testRenderer = renderer.create(
      <Modal
        isOpen
        onClose={() => {}}
        setVisitors={() => {}}
        title='Título modal'
      />
    );

    try {
      expect(testRenderer.toJSON()).toMatchSnapshot();
    } finally {
      testRenderer.unmount();
    }
  });

  it('Should not render', () => {
    const testRenderer = renderer.create(
      <Modal
        isOpen={false}
        onClose={() => {}}
        setVisitors={() => {}}
        title='Título modal'
      />
    );

    try {
      expect(testRenderer.toJSON()).toMatchSnapshot();
    } finally {
      testRenderer.unmount();
    }
  });

  it('Should create when visitorUuid prop is null', () => {
    const { getByText, getByTestId } = render(
      <Modal
        isOpen
        onClose={() => {}}
        setVisitors={() => {}}
        visitors={[]}
        title='Título modal'
      />
    );

    fireEvent.change(getByTestId(/name-input/i), { target: { value: 'João' } });
    fireEvent.change(getByTestId(/rg-input/i), { target: { value: '0123456789' } });
    fireEvent.change(getByTestId(/cpf-input/i), { target: { value: '01201201201' } });
    fireEvent.change(getByTestId(/phone-input/i), { target: { value: '51 998565215' } });
    fireEvent.change(getByTestId(/gender-select/i), { target: { value: 'M' } });
    fireEvent.change(getByTestId(/relation-select/i), { target: { value: 'parente' } });
    fireEvent.change(getByTestId(/permission-select/i), { target: { value: 1 } });

    fireEvent.click(getByText(/Salvar/i));

    expect(mockAddDocFunc).toHaveBeenCalled();
  });
});
