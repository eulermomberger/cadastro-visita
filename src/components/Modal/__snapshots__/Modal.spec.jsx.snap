// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component Should not render 1`] = `<div />`;

exports[`Modal component Should renders correctly 1`] = `
<div>
  <div
    className="popup"
  >
    <div
      className="popup-content"
    >
      <div
        className="popup-header"
      >
        <h2
          className="popup-title"
        >
          Título modal
        </h2>
        <span
          className="close-icon"
          onClick={[Function]}
        >
          ✖
        </span>
      </div>
      <div
        className="popup-body"
      >
        <div
          className="row"
        >
          <input
            className=""
            data-testid="name-input"
            onChange={[Function]}
            placeholder="Nome"
            type="text"
            value=""
          />
        </div>
        <div
          className="row"
        >
          <input
            className=""
            data-testid="rg-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onPaste={[Function]}
            placeholder="RG"
            type="text"
            value=""
          />
          <input
            className=""
            data-testid="cpf-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onPaste={[Function]}
            placeholder="CPF"
            type="text"
            value=""
          />
        </div>
        <div
          className="row"
        >
          <input
            className=""
            data-testid="phone-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onPaste={[Function]}
            placeholder="Telefone"
            type="text"
            value=""
          />
          <select
            data-testid="gender-select"
            onChange={[Function]}
            value=""
          >
            <option
              disabled={true}
              value=""
            >
              Gênero
            </option>
            <option
              value="M"
            >
              Masculino
            </option>
            <option
              value="F"
            >
              Feminino
            </option>
            <option
              value={null}
            >
              Outro
            </option>
          </select>
        </div>
        <div
          className="row"
        >
          <select
            data-testid="relation-select"
            onChange={[Function]}
            value=""
          >
            <option
              disabled={true}
              value=""
            >
              Relação com o morador
            </option>
            <option
              value="parente"
            >
              Parente
            </option>
            <option
              value="funcionario"
            >
              Funcionário
            </option>
            <option
              value="outro"
            >
              Outro
            </option>
          </select>
        </div>
        <div
          className="row"
        >
          <select
            data-testid="permission-select"
            onChange={[Function]}
            value=""
          >
            <option
              disabled={true}
              value=""
            >
              Permissão de entrada
            </option>
            <option
              value={0}
            >
              Bloqueado
            </option>
            <option
              value={1}
            >
              Todos os dias
            </option>
          </select>
        </div>
        <div
          className="row-checkbox"
        >
          <label
            className="white-label"
            onClick={[Function]}
          >
            Possui carro
          </label>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
      <div
        className="popup-footer"
      >
        <button
          className="button-container"
          onClick={[Function]}
          style={
            {
              "backgroundColor": "#2D42FF",
            }
          }
        >
          Salvar
        </button>
      </div>
    </div>
  </div>
</div>
`;
